src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -13288,6 +13079,114 @@ private void LeaveQuery()
             _syntaxFactoryContext.QueryDepth--;
         }
 
+        private delegate (SyntaxToken openToken, PostSkipAction action) SkipBadTokens<TNode>(
+            LanguageParser parser, SyntaxToken openToken, SeparatedSyntaxListBuilder<TNode> builder, SyntaxKind expectedKind, SyntaxKind closeTokenKind) where TNode : GreenNode;
+
+        private SeparatedSyntaxList<TNode> ParseCommaSeparatedSyntaxList<TNode>(
+            ref SyntaxToken openToken,
+            SyntaxKind closeTokenKind,
+            Func<LanguageParser, bool> isPossibleElement,
+            Func<LanguageParser, TNode> parseElement,
+            SkipBadTokens<TNode> skipBadTokens,
+            bool allowTrailingSeparator,
+            SyntaxKind expectedKind = SyntaxKind.IdentifierToken,
```

Comment:
all but one caller passes in `SyntaxKind.IdentifierToken` by default here.  I'd like to just remove this and have thsi be baked in.  But it would change one caller, so it would change a few tests.  Want to hear from compiler team i that's ok or not.